code/modules/recycling/conveyor_filter_advanced.dm

```diff
@@ -0,0 +1,141 @@
+/obj/machinery/conveyor/filter_advanced
+	icon = 'icons/obj/recycling.dmi'
+	icon_state = "filter_generic_off" // todo: sprite or something ah
+	name = "advanced conveyor filter"
+	desc = "A filter that checks for specific items."
+	var/filter_object_name = "Nothing"
+	var/filter_typepath = null
+	var/filter_icon_path = null
+	var/filter_icon_state = null
+	var/filter_output_dir = SOUTH
+	var/unfiltered_output_dir = EAST
+
+/obj/machinery/conveyor/filter_advanced/Initialize()
+	update_desc()
+	. = ..()
+	
+/obj/machinery/conveyor/filter_advanced/update_icon_state()
+	cut_overlays()
+	add_overlay(icon('icons/obj/conveyor_filters.dmi', "filter_[filter_output_dir]"))
+	add_overlay(icon('icons/obj/conveyor_filters.dmi', "unfiltered_[unfiltered_output_dir]"))
+	if(filter_typepath)
+		var/image/I = image(icon(filter_icon_path, filter_icon_state))
+		I.transform *= 0.5
+		add_overlay(I)
+
+
+/obj/machinery/conveyor/filter_advanced/proc/update_desc()
+	desc = "A filter that checks for specific objects and redirects them in another direction.\nMatches get sent [dir2text(filter_output_dir)], everything else goes [dir2text(unfiltered_output_dir)].\nA screwdriver changes the output direction of filtered items, while a wrench changes the output direction of everything else."
+	if(!filter_typepath)
+		desc += "\nIt currently has no set filter! The first object to move onto it will be filtered."
+	else:
+		desc += "\nIt is set to filter out [filter_object_name]. The filter can be cleared with an empty hand."
+
+/obj/machinery/conveyor/filter_advanced/proc/get_it_twisted(var/angle)
+	switch(angle)
+		if(NORTH)
+			return EAST
+		if(EAST)
+			return SOUTH
+		if(SOUTH)
+			return WEST
+	return NORTH
+
+/obj/machinery/conveyor/filter_advanced/convey(list/affecting)
+	for(var/am in affecting)
+		if(!ismovable(am))
+			continue
+		var/atom/movable/movable_thing = am
+		stoplag()
+		if(QDELETED(movable_thing) || (movable_thing.loc != loc))
+			continue
+		if(iseffect(movable_thing) || isdead(movable_thing))
+			continue
+		if(isliving(movable_thing))
+			var/mob/living/zoommob = movable_thing
+			if((zoommob.movement_type & FLYING) && !zoommob.stat)
+				continue
+		if(!movable_thing.anchored && movable_thing.has_gravity())
+			if(!filter_typepath) // Set the filter to the first object that moves over it
+				if(istype(movable_thing, /mob/living/carbon/human)) // Otherwise it'll use the name of the person who walks over it
+					filter_object_name = "people"
+					filter_icon_state = "human_basic"
+				else
+					filter_object_name = movable_thing.name
+					filter_icon_state = movable_thing.icon_state
+				filter_typepath = movable_thing.type
+				filter_icon_path = movable_thing.icon
+				update_desc()
+				update_icon_state()
+				playsound(src, 'sound/machines/ping.ogg', 50)
+			if(movable_thing.type == filter_typepath)
+				step(movable_thing, filter_output_dir)
```

Comment:
https://discord.com/channels/1402868984893669397/1403037112160227502/1404218607491088455
This check seems to fail sometimes?

Reply:
yeah got an admin to check, its /obj/item/stack/conveyor